### ISSUE TITLE
preLaunchTask: isWatching & relaunching

### ISSUE THREAD

**Issue by Spown** [Microsoft/vscode#3929]
As suggested by @isidorn in #2086 I'm creating a separate issue. Seems to be relevant o #3669 as well.

So the problem is, if I have a persistent background task that I want to assign as a `preLaunchtask` and...
A. ...**`isWatchnig` isn't set**: the launch command starts only the task so you need hit F5 once more (while the task is running) to actually launch the project. If you terminate the task after the first launch - the new launch will again only start the task.
B. ...**`isWatching` is set to true**: the first launch starts only the task like above, but the second one throws a prompt `"There is an active running task right now. Terminate it first before executing another task."[Terminate/Close]`. you would expect VSCode continue launch attempt after you select to terminate the task or ignore it but no. Whatever you choose you need to launch it again starting the loop of fail.

In my previous issue people seemed to be focused on the flaws of the tasks and only fixed that part. But I'm personally as a user don't so much care if and how tasks are handled, rather I'm annoyed by the fact that every re/launch I get prompted or warned of things not directly relevant to my project. My main complaint is that the launch spawns prompts at all. Every coding session means re/launching the project 3 dozen times. **any interrupts regarding background deamons unless its something critical are breaking the workflow.** For example I don't really care that #2086 was partually fixed because I still get prompted and as a result I turn `preLaunchTask` off completely thus any improvement in this regard is nil for me. So again: There should be an additional setting for either silently restarting the `preLaunchTask` or silently skip the task launch and assuming that the already running one should run until the end of the current session. And/or there should be a default based on the either - in any case prompting is not an option. 

Here is the [test case](/Spown/pltb). **Ideally the `preLaunchTask` (a gulp-watching one) should run only once. Any subsequent re/launch shouldn't prompt me to terminate it** (If I need to I can do it manually at any point) or warn me (just log it somewhere, idc) or anything else. Launch process should only be interrupted by errors caused by the launched project. Background staff shouldn't have the power to interrupt it.

thanx

**Comment by isidorn** [comment]
This issue is not concise enough for me to figure out what needs to be done - I appologize for that.
Closing as this is an old issue and is I believe captured by https://github.com/Microsoft/vscode/issues/17439
